tests/test.sh

```diff
@@ -94,15 +94,15 @@ test_jsn() {
 
   echo "Testing $(basename "$FILENAME") from $URL"
   # jq before 1.6 had a different float->int behavior, so trying to make it consistent in all
-  $CURL "$URL" | jq --sort-keys -e 'walk(if type == "number" then .+0.0 else . end)' > "$FILENAME"
+  $CURL  --dump-header  "$FILENAME.headers" "$URL" | jq --sort-keys -e 'walk(if type == "number" then .+0.0 else . end)' > "$FILENAME"
```

Comment:
Maybe a test like we did under the `martin/tests`, and sure, it's good to test it in `test.sh` either. 

But dump the header for each request it's not a good idea I guess...

Reply:
at first I also thought headers might be too flaky, but than it seems headers only have the relatively stable things... with the possible exception of the length? i.e. in theory we could have a slightly different compression between different brotli implementation versions?  So lets add it, and if we ever run into issues with the test flakiness, we could  filter the header files to remove unstable headers?

Reply:
responding to this explicitely:
- If length changes, then the etag changes as well.
- If encoding changes, then the etag and length changes as well.

We cannot have both tests that
- ensure that the etag does not change and
- not check that the etag changes.

But since we have PRs for this kind of dependency changes, this should be fine.

This is not like the build in main being broken by new cippy lints every few months.